safelife/file_finder.py

```diff
@@ -157,48 +129,142 @@ def safelife_loader(
         If not found, the 'levels' directory will be searched as well.
         If no paths are supplied, this will generate a random level using
         default level generation parameters.
-    repeat : "auto" or bool or int
-        If true, files will be loaded (yielded) repeatedly and forever.
-        If "auto", it repeats if and only if 'paths' points to a single
-        file of procedural generation parameters.
-        If an int, it repeats exactly that many times.
-    shuffle : bool
-        If true, the order of the files will be shuffled.
+    distinct_levels : int
+        Number of distinct levels to produce.
+        If zero, levels are not cached and will be continually regenerated.
+    total_levels : int or None
+        Total number of levels to produce. If negative, levels are looped
+        forever. Defaults to -1 if 'paths' points to a single file of
```

Comment:
This use of -1 to mean infinity feels quite confusing and makes the code hard to read. I suspect it might be cleaner to use either `None` or `np.nan` (though the latter has some pitfalls, you need to use ` is np.nan` to test for it)